### PR TITLE
 log: Null terminate the errors from OpenSSL 

### PR DESCRIPTION
**Pull request by Jakuje**
Should resolve the following oss-fuzz issue:

https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=67807

##### Checklist
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

**Review by xhanulik — APPROVED** [review]

File: src/libopensc/log.c [hunk 1]
```diff
@@ -84,10 +84,21 @@ void sc_do_log_openssl(sc_context_t *ctx, int level, const char *file, int line,
...
+	/* trailing null byte */
+	buffer = malloc(length + 1);
+	if (buffer == NULL) {
+		sc_do_log(ctx, level, file, line, func, "No memory!");
+		goto end;
+	}
+	rc = BIO_read(bio, buffer, length);
+	buffer[length] = '\n';
+	if (rc <= 0) {
+		sc_do_log(ctx, level, file, line, func, "Cannot read OpenSSL error");
```
Comment (frankmorgner) [inline src/libopensc/log.c:101]: I think it would be useful to keep the old error message "Cannot log OpenSSL error"

File: src/libopensc/log.c [hunk 2]
```diff
@@ -84,10 +84,21 @@ void sc_do_log_openssl(sc_context_t *ctx, int level, const char *file, int line,
...
+		goto end;
+	}
+	/* trailing null byte */
+	buffer = malloc(length + 1);
+	if (buffer == NULL) {
+		sc_do_log(ctx, level, file, line, func, "No memory!");
+		goto end;
+	}
+	rc = BIO_read(bio, buffer, length);
+	buffer[length] = '\n';
```
Comment (frankmorgner) [inline src/libopensc/log.c:99]: This is not NUL terminated, this is a newline!

**Review by frankmorgner — CHANGES_REQUESTED** [review]

File: src/libopensc/log.c (hunk 2 above)
Comment (Jakuje) [reply src/libopensc/log.c:99]: :facepalm: will fix!

File: src/libopensc/log.c (hunk 1 above)
Comment (Jakuje) [reply src/libopensc/log.c:101]: The thing was that the same log message was here for different things and I find it useful when they are as specific as possible so one does not have to go through the code and line offsets to figure out which one of these same messages failed. The old message `Cannot log OpenSSL error` is still on line 82 (which I would consider to rewrite also to something about memory issues).

Comment (frankmorgner) [reply src/libopensc/log.c:101]: thanks for the explanation 👍

**Review by frankmorgner — APPROVED** [review]

**Comment by Jakuje** [comment]
Thanks for reviews! Merging to get rid of the spellcheck and fuzzing issues.

**Pull request merged**